Script rest request

PhilippJKU opened the pull request:
#82 & #72 
Formular zur Abgabe und Script zur Fortschrittskontrolle.
Aufruf für Abgabe fehlt noch.

PhilippJKU committed:
Branch für Script und Abgabe via REST

PhilippJKU committed:
Branch für Script und Abgabe via REST

Hinterberger committed:
-GUI design verändert
-Abfragen der eingegebenen Werte im Formular

Hinterberger committed:
- trim() hinzugefügt um die leerzeichen rauszufiltern

Hinterberger committed:
- über con.getErrorStream() kann nun auch die Fehlermedlung angezeigt werden.

PhilippJKU committed:
#72Script_RestRequest
Fortschrittskontrolle fertiggestellt

Hinterberger committed:
- hinzufügen einer combobox um die Sources auswählen zu können

Hinterberger committed:
Merge remote-tracking branch 'origin/Script_RestRequest' into Script_RestRequest

matfa42 commented on src/main/java/com/gepardec/wdg/guiRunner/RestGUI.java:
```diff
@@ -0,0 +1,285 @@
+                    } catch (UnsupportedEncodingException unsupportedEncodingException) {
+                        unsupportedEncodingException.printStackTrace();
+                    } catch (IOException exception) {
+                        exception.printStackTrace();
```
Log/System.out message mit text und excetion cause/message

matfa42 commented on src/main/java/com/gepardec/wdg/guiRunner/RestGUI.java:
```diff
@@ -0,0 +1,285 @@
+import java.net.URL;
+
+public class RestGUI extends JFrame{
+    private JPanel Basic;
```
bitte camel case

Hinterberger committed:
- CamelCase ist nun durchgängig
- Exceptions werden ausgegeben und "geloggt"
- Die fehlermeldung wurde mit zeilenumbrüchen versehen
- xx Combobox mit dem Enum Werten fehlt noch xx

Hinterberger committed:
- Der Combobox bezieht nun ihre Werte von dem Source ENUM
-löschen des ScriptAbgabe.sh

philzelinger commented on src/main/java/com/gepardec/wdg/guiRunner/RestGUI.java:
```diff
@@ -67,83 +77,93 @@ public void actionPerformed(ActionEvent e) {
+                    logger.error(malformedURLException.getMessage(), malformedURLException);
+                    System.out.println(malformedURLException.getMessage());
+                    System.out.println(malformedURLException.getCause());
                     malformedURLException.printStackTrace();
```
Logging expliziter; vernünftige Logmeldung, exception loggen, etc. (siehe backend logging)
Alles system out´s entfernen
alle printstacktrace entfernen

philzelinger commented on src/main/java/com/gepardec/wdg/guiRunner/RestGUI.java:
```diff
@@ -0,0 +1,292 @@
+            public void focusGained(FocusEvent e) {
+                super.focusGained(e);
+                if (jobIdTextField.getText().equalsIgnoreCase("<JobId der Website>")) {
+                    sourceComboBox.setModel(new DefaultComboBoxModel(Source.values()));
```
Das heißt die source combobox wird nur beim klicken auf das textfeld "jobid" befüllt?
Ist sowas nicht anders zu lösen? ^^ wenn nicht, dann so lassen

Hinterberger committed:
- Ändern der Logs
- Entfernen der System.outs
